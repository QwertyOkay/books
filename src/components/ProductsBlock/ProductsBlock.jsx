import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import Section from 'components/Section';
import Container from 'components/Container';
import Header from 'components/Header/Header';
import SortingAndCountingSection from 'components/SortingAndCountingSection/SortingAndCountingSection'; // Подключите новый компонент
import ProductList from 'components/ProductList/ProductList';
import InfiniteScroll from 'components/InfiniteScroll/InfiniteScroll';
import { ProductsContext } from '../../context/ProductsContext';

function ProductsBlock({ type }) {
  const { loadedProducts, isFetchingMore, fetchMoreProducts, setSortKey } =
    useContext(ProductsContext);

  return (
    <Section variant="products">
      <Container variant="ProductsBlock">
        <Header title="Books read this month" />
        <SortingAndCountingSection type={type} onSort={setSortKey} />
        <InfiniteScroll
          onLoadMore={fetchMoreProducts}
          loading={isFetchingMore}
          hasMore={loadedProducts.length < 100}
        >
          <ProductList products={loadedProducts} type={type} />
        </InfiniteScroll>
      </Container>
    </Section>
  );
}

ProductsBlock.propTypes = {
  type: PropTypes.string.isRequired,
};

export default ProductsBlock;
