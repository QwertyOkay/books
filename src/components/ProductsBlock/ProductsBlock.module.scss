@import '../../styles/vars.scss';

.sectionTitle {
  color: var(--1-W1, #fff);
  text-align: left;
  font-family: Inter;
  font-size: 18px;
  font-style: normal;
  font-weight: 600;
  line-height: normal;
  letter-spacing: 0.027px;
  text-transform: capitalize;
  padding: 24px;
  @media screen and (min-width: 1280px) {
    // margin-bottom: 40px;
  }
}

.sectionBox {
  border-bottom: 1px solid #393f45;
}

.sectionHolder {
  display: flex;
  height: 32px;
  padding: 0px 10px 0px 12px;
  align-items: center;
  gap: 10px;

  color: #fff;
  text-align: right;
  font-family: Inter;
  font-size: 18px;
  font-style: normal;
  font-weight: 400;
  line-height: normal;
  letter-spacing: 0.027px;
  text-transform: capitalize;
}

.sectionWrap {
  display: flex;
  align-items: center;
  // padding: 24px 0px 12px;
  padding: 24px 35px 12px;
  justify-content: center;
  // width: 1112px;

  @media screen and (min-width: 1200px) {
    gap: 700px;
  }
}

.sectionNumber {
  color: #fff;
  text-align: right;
  font-family: Inter;
  font-size: 18px;
  font-style: normal;
  font-weight: 400;
  line-height: normal;
  letter-spacing: 0.027px;
  text-transform: capitalize;
  padding: 0px 10px 0px 12px;
}

// .sectionNumberHolder {
//   margin-left: 35px;
// }

.products {
  padding: 12px 32px;
  @media screen and (min-width: 720px) {
    display: flex;
    justify-content: center;
    flex-wrap: wrap;
    width: 100%;
    padding: 0px;
  }
}

.productsItem {
  display: flex;
  justify-content: center;

  @media screen and (min-width: 720px) {
    flex-basis: calc(100% / 2 - 10px * 2);
    margin: 12px;
  }

  @media screen and (min-width: 1280px) {
    flex-basis: calc(3.3333333333%);
  }

  &:not(:last-child) {
    @media screen and (max-width: 719px) {
      margin-bottom: 20px;
    }
  }
}

.holderLoader {
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 0px 0px 25px;
}
