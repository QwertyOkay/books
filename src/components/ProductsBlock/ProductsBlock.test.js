import React from 'react';
import { render } from '@testing-library/react';
import ProductsBlock from 'components/ProductsBlock/ProductsBlock';

beforeAll(() => {
  global.IntersectionObserver = class {
    // eslint-disable-next-line no-useless-constructor
    constructor() {}
    disconnect() {}
    observe() {}
    takeRecords() {}
    unobserve() {}
  };
});

afterAll(() => {
  delete global.IntersectionObserver;
});

describe('ProductsBlock', () => {
  it('renders correctly', () => {
    const mockOnSort = jest.fn();
    const { asFragment } = render(<ProductsBlock type="someType" onSort={mockOnSort} />);
    expect(asFragment()).toMatchSnapshot();
  });
});