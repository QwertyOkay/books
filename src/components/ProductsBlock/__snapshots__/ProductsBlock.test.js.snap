// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductsBlock renders correctly 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div>
        <h2>
          Books read this month
        </h2>
      </div>
      <div>
        <div>
          <span>
            111
          </span>
          <span>
            Books
          </span>
        </div>
        <div>
          <span>
            Sort By
          </span>
          <button
            class="undefined "
          >
            Name
            <span />
          </button>
        </div>
      </div>
      <ul />
      <div />
    </div>
  </div>
</DocumentFragment>
`;
